c2rust-analyze/src/rewrite/expr/convert.rs

```diff
@@ -281,25 +114,10 @@ impl<'a, 'tcx> intravisit::Visitor<'tcx> for HirRewriteVisitor<'a, 'tcx> {
         // the rewrite should occur at the callsite
         let callsite_span = ex.span.source_callsite();
 
-        let locs = self.find_locations(ex);
-        for loc in &locs {
-            self.locations_visited.insert(*loc);
-        }
-
-        let all_rws_unflattened: Vec<_> = locs
-            .iter()
-            .map(|loc| {
-                self.rewrites
-                    .get(loc)
-                    .map(|rws| rws.iter().map(|rw| &rw.kind).collect::<Vec<_>>())
-                    .unwrap_or_else(Vec::new)
-            })
-            .collect();
-
-        let all_rws_unflattened = all_rws_unflattened
-            .iter()
-            .map(|v| v.as_slice())
-            .collect::<Vec<_>>();
+        let mir_rws = self
+            .mir_rewrites
+            .remove(&ex.hir_id)
```

Comment:
Previously (on `self.rewrites`), this was `.get` instead of `.remove`.  If I'm understanding things correctly, that could've been `.remove`, too, right, since `visit_expr` is never called on the same `Expr` in the same visitor pass?  If that's the case, then this is all good.

Reply:
Okay, the `.remove` here seems correct to me then.